src/main/java/me/kavin/piped/server/handlers/auth/UserHandlers.java

```diff
@@ -109,10 +132,277 @@ public static byte[] loginResponse(String user, String pass)
         }
     }
 
-    public static byte[] deleteUserResponse(String session, String pass) throws IOException {
+    public static HttpResponse oidcLoginRequest(OidcProvider provider, String redirectUri) throws Exception {
+        if (StringUtils.isBlank(redirectUri)) {
+            return HttpResponse.ofCode(400).withHtml("redirect is a required parameter");
+        }
+
+        URI callback = new URI(Constants.PUBLIC_URL + "/oidc/" + provider.name + "/callback");
+        CodeVerifier codeVerifier = new CodeVerifier();
+        OidcData data = new OidcData(redirectUri, codeVerifier);
+        String state = data.getState();
+
+        DatabaseHelper.setOidcData(data);
+
+        AuthenticationRequest oidcRequest = new AuthenticationRequest.Builder(
+                new ResponseType("code"),
+                new Scope("openid"),
+                provider.clientID, callback
+        )
+                .endpointURI(provider.authUri)
+                .codeChallenge(codeVerifier, CodeChallengeMethod.S256)
+                .state(new State(state))
+                .nonce(data.getOidNonce()).build();
+
+        if (redirectUri.equals(Constants.FRONTEND_URL + "/login")) {
```

Comment:
We probably shouldn't hardcode the url path here, maybe it makes sense to parse the domain of the `redirectUri` and see if equals `Constants.FRONTEND_URL`? Or we could just do a `redirectUri.startsWith(Constants.FRONTEND_URL + "/")`.